### PR TITLE
feat(tooltip): Support of v14 tooltip

### PR DESCRIPTION
**Pull request by EmiBemi**

### CODE REVIEW

File: src/tooltip/foundation.tsx [hunk 1]
```diff
@@ -19,59 +20,52 @@ export const useToolTipFoundation = (
...
           return rootEl.ref?.parentElement?.getBoundingClientRect() ?? null;
         },
         getAnchorAttribute: (attr) => {
-          return anchorEl.ref
-            ? (anchorEl.ref?.getAttribute(attr) as string | null)
-            : null;
-        },
-        setAnchorAttribute: (attr, value) => {
-          anchorEl?.setProp(attr as any, value);
+          return anchorEl.ref ? anchorEl.ref?.getAttribute(attr) : null;
```
Comment (jensborch) [inline src/tooltip/foundation.tsx:58]: Can't you write this as 

```
anchorEl.ref?.getAttribute(attr) ?? null
```

as above

**Review by jensborch — CHANGES_REQUESTED** [review]
One very small comment :-)

**Review by jensborch — APPROVED** [review]

File: src/tooltip/index.tsx [hunk 2]
```diff
@@ -3,14 +3,28 @@ import * as RMWC from '@rmwc/types';
 import { classNames, createComponent, Tag } from '@rmwc/base';
 import { useProviderContext } from '@rmwc/provider';
 import { useToolTipFoundation } from './foundation';
-import { AnchorBoundaryType, CssClasses } from '@material/tooltip';
+import { AnchorBoundaryType } from '@material/tooltip';
 
 export type TooltipActivationT = 'hover' | 'click' | 'focus';
+export type TooltipAlignT =
```
Comment (bjarkehs) [inline src/tooltip/index.tsx:9]: If you move this to a shared types file, then you can import it an use it in `constants.ts`.

File: src/tooltip/constants.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
+export const ALIGN_MAP: {
```
Comment (bjarkehs) [inline src/tooltip/constants.ts:1]: If you rewrite this as: 
```typescript
export const ALIGN_MAP: Record<TooltipAlignT,{ xPos?: number; yPos?: number; withCaretPos?: number }>
```

Then you cannot accidentally forget a key.

File: src/tooltip/index.tsx [hunk 4]
```diff
@@ -23,12 +37,14 @@ export interface TooltipProps {
   leaveDelay?: number;
   /** Manually control the open state */
   open?: boolean;
-  /** Whether or not to show an arrow on the Tooltip. Defaults to `false`. */
-  showArrow?: boolean;
```
Comment (bjarkehs) [inline src/tooltip/index.tsx:27]: Is this no longer possible to support or?

File: src/tooltip/readme.tsx [hunk 5]
```diff
@@ -138,6 +156,26 @@ export default function Readme() {
         </>
       </DocsExample>
 
+      <DocsExample label="Alignment">
+        <>
+          <Tooltip overlay={'Align start'} align="start">
```
Comment (bjarkehs) [inline src/tooltip/readme.tsx:161]: [nit] The outer `{` is not necessary? So you could write these as `overlay="Align start"`

File: src/tooltip/index.tsx (hunk 4 above)
Comment (EmiBemi) [reply src/tooltip/index.tsx:27]: I can't really find any documentation for it.. but MCW do expose some methods for controlling a "caret". However, I think it is only for rich tooltips and there is no documentation (that I can find)

**Review by jensborch — APPROVED** [review]

**Pull request merged**